src/test/java/net/minecraftforge/debug/client/model/NewModelLoaderTest.java

```diff
@@ -0,0 +1,102 @@
+import net.minecraft.entity.Entity;
+import net.minecraft.inventory.EquipmentSlotType;
+import net.minecraft.item.*;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.client.event.RenderGameOverlayEvent;
+import net.minecraftforge.client.model.ModelLoader;
+import net.minecraftforge.client.model.obj.OBJLoader;
+import net.minecraftforge.client.model.obj.OBJModel;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.common.util.NonNullLazy;
+import net.minecraftforge.eventbus.api.IEventBus;
+import net.minecraftforge.fml.RegistryObject;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import net.minecraftforge.registries.DeferredRegister;
+import net.minecraftforge.registries.ForgeRegistries;
+
+import javax.annotation.Nullable;
+
+@Mod(NewModelLoaderTest.MODID)
+public class NewModelLoaderTest
+{
+    public static final String MODID = "new_model_loader_test";
+    public static final DeferredRegister<Block> BLOCKS = new DeferredRegister<>(ForgeRegistries.BLOCKS, MODID);
+    public static final DeferredRegister<Item> ITEMS = new DeferredRegister<>(ForgeRegistries.ITEMS, MODID);
+
+    public static RegistryObject<Block> obj_block = BLOCKS.register("obj_block", () ->
+            new Block(Block.Properties.create(Material.WOOD)) {
+                @Override
+                protected void fillStateContainer(StateContainer.Builder<Block, BlockState> builder)
+                {
+                    builder.add(BlockStateProperties.HORIZONTAL_FACING);
+                }
+
+                @Nullable
+                @Override
+                public BlockState getStateForPlacement(BlockItemUseContext context)
+                {
+                    return getDefaultState().with(
+                            BlockStateProperties.HORIZONTAL_FACING, context.getPlacementHorizontalFacing()
+                    );
+                }
+            }
+    );
+
+    public static RegistryObject<Item> obj_item = ITEMS.register("obj_block", () ->
+            new BlockItem(obj_block.get(), new Item.Properties().group(ItemGroup.MISC)) {
+                @Override
+                public boolean canEquip(ItemStack stack, EquipmentSlotType armorType, Entity entity)
+                {
+                    return armorType == EquipmentSlotType.HEAD;
+                }
+            }
+    );
+
+    public static RegistryObject<Item> custom_transforms = ITEMS.register("custom_transforms", () ->
+            new Item(new Item.Properties().group(ItemGroup.MISC))
+    );
+
+    public static RegistryObject<Item> custom_vanilla_loader = ITEMS.register("custom_vanilla_loader", () ->
+            new Item(new Item.Properties().group(ItemGroup.MISC))
+    );
+
+    public NewModelLoaderTest()
```

Comment:
Shouldn't this also test adding a custom loader?

Reply:
Possibly, yeah. Actually just looking at it brought something to my attention: we might run into race conditions if people call registerLoader from the FMLClientSetupEvent. I'll make the loader registry a concurrent map while at it.